### ISSUE TITLE
uint64 use

### ISSUE THREAD

**Issue by sdboyer** [donovanhide/stockfighter#1]
Hi - 

You opted for `uint64` as the type for most of the integer values in the API. I'm curious if there was a specific reason to use that, rather than `int64` (or even just `int`)? Clearly this is a minor nit, but I do have to convert to `int64` prior to passing data to influxdb, which is awkward. So I figured I'd ask :smile:

FWIW, if I had to stake a bet on whether they'd try to fuck with our bots by overflowing an `int64` or negatives, I'd guess the latter. (Being as strict as possible wrt what JSON allows would be a `float64`, I think.)

**Comment by donovanhide** [comment]
I guess I just wanted to be alerted if a price or quantity was supplied as an illogical negative:

http://play.golang.org/p/sImKeZkjxc

Probably having a type alias with custom Marshal functions which error on bad inputs would be the safest. If you want to do a patch with:

`type Number int64`

which errors on decoding JSON negative values, I'll happily merge it :-)

**Comment by sdboyer** [comment]
Sweet. Yeah, I think that's an important case to capture.

...Though really, it seems like there are maybe a lot of potentially important things to capture, and given what the API docs say about not being able to rely on the API docs, or the consistency of responses, maybe it's better to take a general approach to validating all incoming data, as well as checking for missing or additional fields?

If that's a worthwhile approach, then maybe a subtype isn't the best idea; such information would be better reported in a (stack of) errors, and an `UnmarshalJSON` implementation, possibly even relying on a `json.Decoder`...

Hmm. Starting to seem like more work than it's worth, perhaps. I need to ponder on it.

**Comment by donovanhide** [comment]
I guess we only know what is needed when we've got to the last level :-)

**Comment by sdboyer** [comment]
Yeah, that seems like part of the tricky bit - as written right now, I think it'll just hide missing struct values, or drop values not present in the struct - silently. No way to know it's happening.

**Comment by donovanhide** [comment]
You can use a `debug` flag to compare the JSON to the structs in your own code. Eg:

https://github.com/donovanhide/stockfighter/blob/master/api_test.go#L10

https://github.com/donovanhide/stockfighter/blob/master/api_test.go#L53

**Comment by sdboyer** [comment]
yep, i used that when i was first getting set up. it didn't strike me as practical for discoverin the occasional erroneous outlier. catching those automatically, and notifying myself via a more-structured error, was the approach i was considering.

seems like overkill for right now, now that i write it out :smile:
